universum/lib/utils.py

```diff
@@ -79,18 +95,18 @@ def create_driver(local_factory, teamcity_factory, jenkins_factory, env_type="")
     return local_factory()
 
 
-def format_traceback(exc, trace):
-    tb_lines = traceback.format_exception(exc.__class__, exc, trace)
-    tb_text = ''.join(tb_lines)
+def format_traceback(exc: Exception, trace: Optional[TracebackType]) -> str:
+    tb_lines: List[str] = traceback.format_exception(exc.__class__, exc, trace)
+    tb_text: str = ''.join(tb_lines)
     return tb_text
 
 
-def check_required_option(settings, setting_name, error_message):
+def check_required_option(settings: ModuleSettings, setting_name: str, error_message: str) -> None:
```

Comment:
It should be ModuleNamespace

Reply:
changing ModuleSettings to ModuleNamespace here results in mypy error. Investigating

Reply:
changed to HasModulesMapping Protocol